akka-actor-typed-tests/src/test/java/jdocs/akka/typed/supervision/SupervisionCompileOnlyTest.java

```diff
@@ -135,4 +133,33 @@ public void supervision() {
   }
   // #restart-keep-children
 
+  interface Resource {
+    void close();
+    void process(String[] parts);
+  }
+  public static Resource claimResource() {
+    return null;
+  }
+
+  static void prerestartBehavior() {
+    //#restart-PreRestart-signal
+    Behaviors.supervise(
+      Behaviors.<String>setup(ctx -> {
+        final Resource resource = claimResource();
+
+        return Behaviors.receive((notUsed, msg) -> {
+          // there might be bugs here...
```

Comment:
I think on these comments I mean, is there something clearer we can say?

Reply:
Changed to ` // message handling that might throw an exception` to make it more clear